### PR TITLE
Doc updates

### LINKED ISSUE #1056: Documentation about module use
Let's add a note right before or right after the table of systems in https://spack-stack.readthedocs.io/en/latest/PreConfiguredSites.html that makes clear that the preferred way to use them is 'module use <whatever path>/install/modulefiles/Core'. I feel like I get this question a lot, so let's make the documentation as unambiguous as possible. I'll add this when I get a chance.

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
### Summary

This PR updates readthedocs:
- Create table to track add-on envs and make them easy to find.
- Clarify how to use the pre-configured sites table.
- Change 'Gaea C5' -> 'Gaea'

### Testing

Tested doc build on personal machine.

### Applications affected

n/a

### Systems affected

n/a

### Dependencies

none

### Issue(s) addressed

Fixes #1056

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Comment by AlexanderRichert-NOAA** [comment]
@climbfuji this can get merged before or after the 1.7.0->develop merge.

**Comment by climbfuji** [comment]
Let's hold off, there'll be multiple merge conflicts and chances are that I will revert some of your updates when I merge the release branch back.

**Comment by climbfuji** [comment]
@AlexanderRichert-NOAA I updated your PR from develop and fixed the merge conflicts. See 22172a0.

The documentation renders correctly I think: https://spack-stack--1059.org.readthedocs.build/en/1059/PreConfiguredSites.html

Please take a look at it and merge if you are happy with it, I am going to approve it now. Thanks for making those updates.

**Review by climbfuji — APPROVED** [review]

File: doc/source/PreConfiguredSites.rst [hunk 1]
```diff
@@ -413,6 +432,8 @@ NOAA Acorn (WCOSS2 test system)
 
 For spack-stack-1.7.0, the meta modules are in ``/lfs/h1/emc/nceplibs/noscrub/spack-stack/spack-stack-1.7.0/envs/ue-intel{19,2022}/modulefiles/Core``.
 
+Acorn has a spack-stack environment installed to reflect the versions and build options used on WCOSS2 for the operational configurations of the UFS Weather Model. The path to that environment is ``/lfs/h1/emc/nceplibs/noscrub/spack-stack/spack-stack-1.6.0/envs/ufswm-env``.
+
```
Comment (AlexanderRichert-NOAA) [inline doc/source/PreConfiguredSites.rst:436]: Let's remove this since it's redundant with the new table up above.

**Comment by AlexanderRichert-NOAA** [comment]
Thanks! In the first sentence of the 'Supplemental environments' section, 'environments are maintained' should be 'environments that are maintained', and I made one other comment in the Acorn section, but otherwise looks good.

**Comment by AlexanderRichert-NOAA** [comment]
@climbfuji thanks!

**Pull request merged**